extension/manifest/_manifest.py

```diff
@@ -0,0 +1,206 @@
+            + self.padding_size.to_bytes(4, byteorder=_MANIFEST_BYTEORDER)
+            # uint64_t: Size in bytes between the manifest and the data it was appended to.
+            + self.program_offset.to_bytes(8, byteorder=_MANIFEST_BYTEORDER)
+            # uint32_t: Actual size of this manifest.
+            + self.length.to_bytes(4, byteorder=_MANIFEST_BYTEORDER)
+            # Manifest magic. This lets consumers detect whether the
+            # manifest was inserted or not. Always use the proper magic value
+            # (i.e., ignore self.magic) since there's no reason to create an
+            # invalid manifest.
+            + self.EXPECTED_MAGIC
+        )
+        return data
+
+    @staticmethod
+    def from_bytes(data: bytes) -> "_ManifestLayout":
+        """Tries to read a manifest from the provided data.
+
+        Does not validate that the header is well-formed. Callers should
+        use is_valid().
+
+        Args:
+            data: The data to read from.
+        Returns:
+            The contents of the serialized manifest.
+        Raises:
+            ValueError: If not enough data is provided.
+        """
+        if len(data) <= _ManifestLayout.EXPECTED_MIN_LENGTH:
+            raise ValueError(
+                f"Not enough data for the manifest: {len(data)} "
+                + f"< {_ManifestLayout.EXPECTED_MIN_LENGTH}"
+            )
+        magic = data[-4:]
+        length = int.from_bytes(data[-8:-4], byteorder=_MANIFEST_BYTEORDER)
+        program_offset = int.from_bytes(data[-16:-8], byteorder=_MANIFEST_BYTEORDER)
+        padding_size = int.from_bytes(data[-20:-16], byteorder=_MANIFEST_BYTEORDER)
+        signature_size = int.from_bytes(data[-24:-20], byteorder=_MANIFEST_BYTEORDER)
+        signature = data[-(signature_size + 24) : -24]
+        return _ManifestLayout(
+            signature=signature,
+            signature_size=signature_size,
+            padding_size=padding_size,
+            program_offset=program_offset,
+            length=length,
+            magic=magic,
+        )
+
+    @staticmethod
+    def from_manifest(manifest: "Manifest") -> "_ManifestLayout":
+        return _ManifestLayout(
+            signature=manifest.signature,
+            signature_size=len(manifest.signature),
+            length=_ManifestLayout.EXPECTED_MIN_LENGTH + len(manifest.signature),
+            # program_offset and padding_size are set at append time.
+        )
+
+
+@experimental("This API is experimental and subject to change without notice.")
+@dataclass
+class Manifest:
+    """A manifest that can be appended to a binary blob. The manifest contains
+    meta information about the binary blob. You must know who created the manifest
+    to be able to interpret the data in the manifest."""
+
+    # Unique ID for the data the manifest was appended to. Often this might contain
+    # a crytographic signature for the data.
+    signature: bytes
+
```

Comment:
Add version

Reply:
I can imagine people can use this for other use-cases besides security, such as saving arbitrary serializable metadata.

For instance, saving tokenizer.json file location etc.